packages/python/src/armonik/common/filter.py

```diff
@@ -0,0 +1,443 @@
+from __future__ import annotations
+from typing import List, Any, Type, Union
+from google.protobuf.message import Message
+import google.protobuf.timestamp_pb2 as timestamp
+from ..protogen.common.filters_common_pb2 import *
+
+
+class Filter:
+    """
+    Base class for all filters
+    """
+    def message_type(self) -> Type:
```

Comment:
Use proper decorators and classes to say this is an abstract class. Do not re-implement the wheel